StackExchange.Redis/StackExchange/Redis/RedisDatabase.cs

```diff
@@ -2164,6 +2805,171 @@ private Message GetSortedSetRemoveRangeByScoreMessage(RedisKey key, double start
                     GetRange(start, exclude, true), GetRange(stop, exclude, false));
         }
 
+        private Message GetStreamAcknowledgeMessage(RedisKey key, RedisValue groupName, CommandFlags flags, params string[] messageIds)
+        {
+            var arr = new RedisValue[messageIds.Length + 2];
+
+            var offset = 0;
+
+            arr[offset++] = key.AsRedisValue();
+            arr[offset++] = groupName;
+
+            for (var i = 0; i < messageIds.Length; i++)
+            {
+                arr[offset++] = messageIds[i];
+            }
+
+            return Message.Create(Database, flags, RedisCommand.XACK, arr);
+        }
+
+        private Message GetStreamAddMessage(RedisKey key, RedisValue entryId, int? maxLength, bool useApproximateMaxLength, CommandFlags flags, params NameValueEntry[] streamFields)
+        {
+            // See https://redis.io/commands/xadd.
+
+            if (streamFields == null) throw new ArgumentNullException(nameof(streamFields));
+            if (streamFields.Length == 0) throw new ArgumentOutOfRangeException("streamFields must contain at least one field.");
+
+            if (maxLength.HasValue && maxLength <= 0)
+            {
+                throw new ArgumentOutOfRangeException("maxLength must be greater than 0.");
```

Comment:
`throw new ArgumentOutOfRangeException(nameof(maxLength), "maxLength must be greater than 0.");`